CONTRIBUTING.md

```diff
@@ -0,0 +1,158 @@
+# Contributing
+
+Thanks for checking out Clay. We accept pull requests from anybody, however we ask
+that you follow some conventions when contributing to Clay.
+
+We have set up a
+[piece-of-cake](https://github.com/pebble/clay/issues?q=is%3Aopen+is%3Aissue+label%3Apiece-of-cake)
+label in GitHub issues for bugs/features that are easy to implement. These issues are
+a great place to start for people that are new to the codebase.
+
+
+## Requirements
+
+ - [Node](https://nodejs.org) (4.0 and later)
+ - [Google Chrome](www.google.com/chrome) (To run tests)
+ - [Pebble SDK](https://developer.pebble.com/sdk/) (optional)
+
+
+## Getting Started
+
+### 1. Fork, then clone the repo:
+
+`$ git clone git@github.com:your-username/clay.git`
+
+### 2. Create a new branch with the following naming convention.
+
+`#<issue-id>/description-of-change`
+
+example:
+
+`#90/support-message-keys`
+
+If there is currently no GitHub issue open for your proposed contribution,
+then please make one. This allows us to assign milestones and triage priority.
+
+### 3. Install the project dependencies.
+
+`$ npm install`
+
+### 4. Develop your feature
+
+Commit as often as you like, however please squash commits that fix typos or are
+otherwise noisy.
+
+### 5. Run the tests
+
+`$ npm run test`
+
+### 6. Run linting
+
+`$ npm run lint`
+
+### 7. Submit your pull request
+
+In the main comment of the pull request, you must state the GitHub issue that your PR
+resolves. You do this by using the syntax `Resolves #<issue-id>`. An example of this
+can be found [here](https://github.com/pebble/clay/pull/81)
+
+
+
+# Project Structure and Development
+
+
+## Important Files
+
+There are two main entry points for Clay - `index.js` and `src/scripts/config-page.js`.
+
+### index.js
+
+This is the main entry point for the code that will run in the Pebble app's `src/js/app.js`.
+It is responsible for serializing the provided config into a data URI that will be opened
+using `Pebble.openURL()`. It also persists data to `localStorage`.
+
+### src/scripts/config-page.js
+
+This is the main entry point for the code that runs on the generated config page,
+and is responsible for passing the injected config and other components to the
+`ClayConfig` class.
+
+
+## Building
+
+There are two ways to build Clay, production mode and development mode.
```

Comment:
Might be good to explain the differences and why/when developers should use each mode